In [ ]:
import pandas as pd
import numpy as np
import matplotlib as plt
test_logs = pd.read_csv(r"C:\Users\SRIVIDHYA\Desktop\ArkoseLabs\arkose_labs_DS_test_logs.csv")
test_logs.head()



The file is  extracted as csv to a dataframe and initial view to file shows there are 'Nan" values for the columns "Unnamed 4 to 29".So drop those columns for better analysis of the data set.So the new data frame has only 4 columns that is needed for our analysis.


In [2]:
new_test_log = test_logs.drop(test_logs.loc[:,'Unnamed: 4':'Unnamed: 29'].head(0).columns, axis=1)


In [3]:

new_test_log.shape

(1779, 4)

The data frame has 4 columns and 1779 rows .The column 'solved' has null values to be handled.The message and solved are related like if the message is "user submitted answer' is '1' for solved  for other messages it is 'Nan'.So only if the user submitted correct answer is needed to  find the completion rate.so we can fill the other message columns as '0' since only the right answers  with user submitted are going to be considered for the completion rate.

In [4]:

new_test_log.isnull().sum()

category       0
message        0
solved      1295
theme_ab       0
dtype: int64

In [5]:
new_test_log.head(30)

category                message  solved  theme_ab
0   check answer  user submitted answer     0.0         2
1         loaded            game loaded     NaN         1
2           user    user clicked verify     NaN         2
3   check answer  user submitted answer     0.0         1
4         loaded            game loaded     NaN         1
5   check answer  user submitted answer     1.0         1
6   check answer  user submitted answer     1.0         2
7         loaded            game loaded     NaN         1
8           user    user clicked verify     NaN         1
9   check answer  user submitted answer     0.0         1
10          user         user refreshed     NaN         1
11        loaded            game loaded     NaN         1
12        loaded            game loaded     NaN         2
13        loaded            game loaded     NaN         1
14          user    user clicked verify     NaN         1
15  audio answer  user submitted answer     1.0         1
16        loaded            game loaded     NaN         2
17          user    user clicked verify     NaN         1
18          user    user clicked verify     NaN         1
19        loaded            game loaded     NaN         1
20        loaded            game loaded     NaN         2
21          user    user clicked verify     NaN         2
22        loaded            game loaded     NaN         2
23        loaded            game loaded     NaN         1
24        loaded            game loaded     NaN         2
25  check answer  user submitted answer     1.0         1
26          user    user clicked verify     NaN         1
27        loaded            game loaded     NaN         1
28  check answer  user submitted answer     1.0         2
29        loaded            game loaded     NaN         1

In [6]:
new_test_log.loc[new_test_log["message"] == "game loaded", "solved"] = 0.0
new_test_log.loc[new_test_log["message"] == "user clicked verify", "solved"] = 0.0
new_test_log.loc[new_test_log["message"] == "user refreshed", "solved"] = 0.0


So now the data set is clean and ready to find the completion rate of A and B with respect to correct answers.

In [7]:
print(new_test_log['solved'].value_counts(dropna = False))

0.0    1418
1.0     361
Name: solved, dtype: int64


In [59]:
new_test_log.head(30)

category                message  solved  theme_ab
0   check answer  user submitted answer       0         2
1         loaded            game loaded       0         1
2           user    user clicked verify       0         2
3   check answer  user submitted answer       0         1
4         loaded            game loaded       0         1
5   check answer  user submitted answer       1         1
6   check answer  user submitted answer       1         2
7         loaded            game loaded       0         1
8           user    user clicked verify       0         1
9   check answer  user submitted answer       0         1
10          user         user refreshed       0         1
11        loaded            game loaded       0         1
12        loaded            game loaded       0         2
13        loaded            game loaded       0         1
14          user    user clicked verify       0         1
15  audio answer  user submitted answer       1         1
16        loaded            game loaded       0         2
17          user    user clicked verify       0         1
18          user    user clicked verify       0         1
19        loaded            game loaded       0         1
20        loaded            game loaded       0         2
21          user    user clicked verify       0         2
22        loaded            game loaded       0         2
23        loaded            game loaded       0         1
24        loaded            game loaded       0         2
25  check answer  user submitted answer       1         1
26          user    user clicked verify       0         1
27        loaded            game loaded       0         1
28  check answer  user submitted answer       1         2
29        loaded            game loaded       0         1

Select only sorted  column with 1 as values to find the completion rate for A and B


In [36]:
new_test_log['theme_ab'].value_counts()

1    1098
2     681
Name: theme_ab, dtype: int64

In [35]:
if [(new_test_log['solved'] == 1) &  (new_test_log['theme_ab'] == 1)]:
    completion_rate_A = new_test_log['theme_ab'].value_counts(normalize= True).sort_index() * 100
elif [(new_test_log['solved'] == 1) &  (new_test_log['theme_ab'] == 2)]: 
    completion_rate_B = new_test_log['theme_ab'].value_counts(normalize= True).sort_index() * 100
completion_rate_A
completion_rate_B 

1    61.720067
2    38.279933
Name: theme_ab, dtype: float64

The completion rate for users completing the challenge with respect to A is 61.7% and B is 38.27 %.The result of this analysis is in the result_test_log file

In [38]:
new_test_log.to_csv(r"C:\Users\SRIVIDHYA\Desktop\ArkoseLabs\result_test_log.csv")